src/ray/raylet/node_manager.cc

```diff
@@ -1541,6 +1543,7 @@ void NodeManager::DisconnectClient(const std::shared_ptr<ClientConnection> &clie
   }
 
   local_task_manager_->ClearWorkerBacklog(worker->WorkerId());
+  cluster_task_manager_->CancelTaskForOwner(worker->GetAssignedTaskId());
```

Comment:
I think it doesn't clean up if the tasks are queued on other nodes? Can you add a test case?